include/zephyr/net/http/service.h

```diff
@@ -113,6 +146,35 @@ struct http_service_desc {
 				&_CONCAT(_http_resource_desc_##_name, _list_start)[0],             \
 				&_CONCAT(_http_resource_desc_##_name, _list_end)[0])
 
+/**
+ * @brief Define an HTTPS service with static resources.
+ *
+ * @note The @p _host parameter must be non-`NULL`. It is used to specify an IP address either in
+ * IPv4 or IPv6 format a fully-qualified hostname or a virtual host.
+ *
+ * @note The @p _port parameter must be non-`NULL`. It points to a location that specifies the port
+ * number to use for the service. If the specified port number is zero, then an ephemeral port
+ * number will be used and the actual port number assigned will be written back to memory. For
+ * ephemeral port numbers, the memory pointed to by @p _port must be writeable.
+ *
+ * @param _name Name of the service.
+ * @param _host IP address or hostname associated with the service.
+ * @param[inout] _port Pointer to port associated with the service.
+ * @param _concurrent Maximum number of concurrent clients.
+ * @param _backlog Maximum number queued connections.
+ * @param _detail Implementation-specific detail associated with the service.
+ * @param _sec_tag_list TLS security tag list used to setup a HTTPS socket.
+ * @param _sec_tag_list_size TLS security tag list size used to setup a HTTPS socket.
+ */
+#define HTTPS_SERVICE_DEFINE(_name, _host, _port, _concurrent, _backlog, _detail,              \
+			     _sec_tag_list, _sec_tag_list_size)                                \
```

Comment:
I imagine it's an error to use this macro if TLS isn't enabled?

```suggestion
#define HTTPS_SERVICE_DEFINE(_name, _host, _port, _concurrent, _backlog, _detail,              \
			     _sec_tag_list, _sec_tag_list_size)                                \
	BUILD_ASSERT(IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS),                               \
		     "TLS is required for HTTP secure (CONFIG_NET_SOCKETS_SOCKOPT_TLS)");      \
```